app/build.gradle

```diff
@@ -36,7 +40,14 @@ dependencies {
     implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
     implementation 'com.squareup.picasso:picasso:2.71828'
     implementation 'androidx.recyclerview:recyclerview:1.2.0-beta01'
+    implementation platform('com.google.firebase:firebase-bom:26.5.0')
+    implementation 'com.google.firebase:firebase-config:19.1.3'
+    implementation 'com.google.firebase:firebase-analytics:17.2.3'
+    implementation 'com.google.firebase:firebase-crashlytics'
+    implementation 'com.google.firebase:firebase-analytics'
```

Comment:
you have two lines here importing 'com.google.firebase:firebase-analytics'. In one, you have specified version 17.2.3 and in the other you have not, and that probably means "use the latest version". Remove the one without version.

Reply:
I prefer using numbered versions. Otherwise you could make a build with latest version and have everything working. Later I could download your code, build it, and get an even fresher version of the library. Now  when I run the app it might behave differently than when you run the app, even though we have the exact same code. If the version is hardcoded that can not happen.

Reply:
Let me know if you understand what I'm talking about.

Reply:
I understand.